src/samplers/pgibbs.jl

```diff
@@ -143,19 +143,19 @@ end
 assume{T<:Union{PG,SMC}}(spl::Sampler{T}, dist::Distribution, vn::VarName, _::VarInfo) = begin
   vi = current_trace().vi
   if isempty(spl.alg.space) || vn.sym in spl.alg.space
-    vi.index += 1
     if ~haskey(vi, vn)
       r = rand(dist)
       push!(vi, vn, r, dist, spl.alg.gid)
+      addorder!(vi, vn, vi.num_produce)
```

Comment:
I think it's better to actually put the `addorder!()` inside `push!()`. You call `addorder!()` after each `push!()` anyway.